xbmc/pvr/channels/PVRChannelsPath.cpp

```diff
@@ -141,17 +149,18 @@ CPVRChannelsPath::CPVRChannelsPath(bool bRadio, const std::string& strGroupName)
     m_path.append("/");
 }
 
-CPVRChannelsPath::CPVRChannelsPath(bool bRadio, const std::string& strGroupName, const std::string& strClientID, int iChannelUID)
+CPVRChannelsPath::CPVRChannelsPath(bool bRadio, const std::string& strGroupName, const std::string& strClientID, const std::string& instanceID, int iChannelUID)
   : m_bRadio(bRadio)
 {
   if (!strGroupName.empty() && !strClientID.empty() && iChannelUID >= 0)
   {
     m_kind = Kind::CHANNEL;
     m_group = strGroupName;
     m_clientID = strClientID;
+    m_instanceID = instanceID;
     m_iChannelUID = iChannelUID;
-    m_path = StringUtils::Format("pvr://channels/{}/{}/{}_{}.pvr", bRadio ? "radio" : "tv",
-                                 CURL::Encode(m_group), m_clientID, m_iChannelUID);
+    m_path = StringUtils::Format("pvr://channels/{}/{}/{}{}_{}.pvr", bRadio ? "radio" : "tv",
+                                 CURL::Encode(m_group), m_clientID, (m_instanceID.empty() ? "@" + m_instanceID : ""), m_iChannelUID);
```

Comment:
should this read `!m_instanceID.empty()` - seems `!` is missing.

Reply:
Yes you right, has wrongly done also by other places.